cool_seq_tool/data_sources/feature_overlap.py

```diff
@@ -0,0 +1,241 @@
+        """
+        aliases, error_msg = self.seqrepo_access.translate_identifier(
+            identifier, Assembly.GRCH38.value
+        )
+
+        if error_msg:
+            raise FeatureOverlapError(str(error_msg))
+
+        if not aliases:
+            raise FeatureOverlapError(
+                f"Unable to find {Assembly.GRCH38.value} aliases for: {identifier}"
+            )
+
+        assembly_chr_pattern = rf"^{Assembly.GRCH38.value}:(?P<chromosome>{CHR_PATTERN})$"  # noqa: E501
+        for a in aliases:
+            chr_match = re.match(assembly_chr_pattern, a)
+            if chr_match:
+                break
+
+        if not chr_match:
+            raise FeatureOverlapError(
+                f"Unable to find {Assembly.GRCH38.value} chromosome for: {identifier}"
+            )
+
+        chr_groupdict = chr_match.groupdict()
+        return chr_groupdict["chromosome"]
+
+    def get_grch38_mane_gene_cds_overlap(
+        self,
+        start: int,
+        end: int,
+        chromosome: Optional[str] = None,
+        identifier: Optional[str] = None,
+        residue_mode: ResidueMode = ResidueMode.RESIDUE,
+    ) -> Optional[Dict[str, CdsOverlap]]:
+        """Given GRCh38 genomic data, find the overlapping MANE features (gene and cds)
+
+        :param start: GRCh38 start position
+        :param end: GRCh38 end position
+        :param chromosome: Chromosome. 1..22, X, or Y. If not provided, must provide
+            `identifier`. If both `chromosome` and `identifier` are provided,
+            `chromosome` will be used.
+        :param identifier: Genomic identifier on GRCh38 assembly. If not provided, must
+            provide `chromosome`. If both `chromosome` and `identifier` are provided,
+            `chromosome` will be used.
+        :param residue_mode: Residue mode for `start` and `end`
+        :raise FeatureOverlapError: If missing required fields or unable to find
+            associated ga4gh identifier
+        :return: MANE feature (gene/cds) overlap data represented as a dict
+            {
+                gene: {
+                    'cds': VRS Sequence Location
+                    'overlap': VRS Sequence Location
+                }
+            }
+        """
+        ga4gh_seq_id = None
+        if chromosome:
+            if not re.match(f"^{CHR_PATTERN}$", chromosome):
+                raise FeatureOverlapError("`chromosome` must be 1, ..., 22, X, or Y")
+        else:
+            if identifier:
+                chromosome = self._get_chr_from_alt_ac(identifier)
+                if identifier.startswith("ga4gh:SQ."):
+                    ga4gh_seq_id = identifier
+            else:
+                raise FeatureOverlapError(
+                    "Must provide either `chromosome` or `identifier`"
+                )
+
+        # Convert residue to inter-residue
+        if residue_mode == ResidueMode.RESIDUE:
+            if start == end:
+                end -= 1
```

Comment:
If there is a 1 nt search, how is that specified?

Reply:
@ahwagner pushed fix. let me know if this is what you were expecting